exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/ColumnType.kt

```diff
@@ -247,6 +247,66 @@ class EntityIDColumnType<T : Comparable<T>>(
     override fun hashCode(): Int = 31 * super.hashCode() + idColumn.hashCode()
 }
 
+/**
+ * An interface defining the transformation between a database column type and a real type.
+ *
+ * @param TReal The real type to which the column value is transformed.
+ * @param TColumn The column type provided by underlying `ColumnType`.
+ */
+interface ColumnTransformer<TReal : Any, TColumn : Any> {
+    /**
+     * Transforms a column value from the underlying `ColumnType` into the real type.
+     */
+    fun toReal(value: TColumn): TReal
+
+    /**
+     * Transforms a real type value into a column value.
+     */
+    fun toColumn(value: TReal): TColumn
+}
```

Comment:
What would be the plan for this feature in relation to the existing DAO [ColumnWithTransform](https://github.com/JetBrains/Exposed/blob/main/exposed-dao/src/main/kotlin/org/jetbrains/exposed/dao/Entity.kt#L15)? Could that class make use of this transformer instead?

And given how [transform](https://github.com/JetBrains/Exposed/blob/main/exposed-dao/src/main/kotlin/org/jetbrains/exposed/dao/EntityClass.kt#L579) already works for Entity field columns, if the user does not choose to use `memoizedTransform()` for example, could defining the transformation on the table object actually replace defining the transformation on the entity class?

Also, would using the 2 in parallel work ok? For example, defining a column with core/DSL `transform()` in table object. Then defining entity class with a field for that column, which then uses a DAO `transform()` on every read. Would it create similar behavior as in the nested transformations test?

Reply:
Actually, DAO transform and DSL transform are independent.

The main difference is that on the DSL level it's necessary to make column replacement with a new column type. It's needed because during reading/writing values go through the underlying `ColumnType`, so table object must contain the column with the proper column type.

I was trying to make the transform without replacing column but haven't found a way to do that, if you probably have any ideas how it could be done, I'd be happy to know about it). 

From the DAO's perspective transformed DSL column is still just a column, so DAO works as before. I added a test `testDaoTransformWithDslTransform()` to show that DAO and DSL transforms could be mixed.